src/shared/organisms/DataExplorerGraphFlowNavigationStack/NavigationArrows.spec.tsx

```diff
@@ -0,0 +1,238 @@
+  });
+
+  it('should render the back/forward arrows as not disabled', () => {
+    store.dispatch(
+      ResetDataExplorerGraphFlow({ initialState: initialDataExplorerState })
+    );
+    rerender(app);
+
+    const backArrow = container.querySelector('[aria-label="back-arrow"]');
+    const forwardArrow = container.querySelector(
+      '[aria-label="forward-arrow"]'
+    );
+    expect(backArrow).toBeInTheDocument();
+    expect(forwardArrow).toBeInTheDocument();
+  });
+  it('should left side of navigation become 3 and right side become 2 when Forward btn clicked', async () => {
+    store.dispatch(
+      ResetDataExplorerGraphFlow({ initialState: initialDataExplorerState })
+    );
+    rerender(app);
+    const forwardArrow = container.querySelector(
+      '.navigation-arrow-btn[aria-label="forward-arrow"]'
+    );
+    expect(forwardArrow).not.toBeNull();
+    forwardArrow && (await user.click(forwardArrow));
+    rerender(app);
+    expect(store.getState().dataExplorer.leftNodes.links.length).toEqual(3);
+    expect(store.getState().dataExplorer.rightNodes.links.length).toEqual(2);
+  });
+  it('should left side of navigation become 1 and right side become 4 when Back btn clicked', async () => {
+    store.dispatch(
+      ResetDataExplorerGraphFlow({ initialState: initialDataExplorerState })
+    );
+    rerender(app);
+    const backArrow = container.querySelector(
+      '.navigation-arrow-btn[aria-label="back-arrow"]'
+    );
+    expect(backArrow).not.toBeNull();
+    backArrow && (await user.click(backArrow));
+    rerender(app);
+    expect(store.getState().dataExplorer.leftNodes.links.length).toEqual(1);
+    expect(store.getState().dataExplorer.rightNodes.links.length).toEqual(4);
+  });
+  it('should forward btn disappear when there is no more forward navigation', async () => {
+    store.dispatch(
+      ResetDataExplorerGraphFlow({ initialState: initialDataExplorerState })
+    );
+    rerender(app);
+    for (const _ of store.getState().dataExplorer.rightNodes.links) {
+      const forwardArrow = container.querySelector(
+        '.navigation-arrow-btn[aria-label="forward-arrow"]'
+      );
+      expect(forwardArrow).not.toBeNull();
+      forwardArrow && (await user.click(forwardArrow));
+      rerender(app);
+    }
+    expect(store.getState().dataExplorer.leftNodes.links.length).toEqual(5);
+    expect(store.getState().dataExplorer.rightNodes.links.length).toEqual(0);
+    const forwardArrowAfterFullNavigation = container.querySelector(
+      '.navigation-arrow-btn[aria-label="forward-arrow"]'
+    );
+    expect(forwardArrowAfterFullNavigation).toBeNull();
+  });
+  it('should return to /my-data when there is no more back navigation', async () => {
+    store.dispatch(
+      ResetDataExplorerGraphFlow({ initialState: initialDataExplorerState })
+    );
+    rerender(app);
+    for (const _ of store.getState().dataExplorer.leftNodes.links) {
+      const backArrow = container.querySelector(
```

Comment:
Since we are querying the back and forward buttons, so often in these tests, it might be a good idea to create a tiny utility for it:

```typescript
const getButton = (type: 'forward' | 'back') => container.querySelector(`.navigation-arrow-btn[aria-label=${type}-button]`);
```